extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java

```diff
@@ -108,9 +116,17 @@ public IntBuffer readKeyValues(IntBuffer keyDest, int nullPlaceholder) throws IO
         }
     }
 
+    /**
+     * If {@link #pageHeader} is {@code null}, read it from the file and increment the {@link #offset} and file position
+     * by the length of page header. This method assumes that file position is set to {@link #offset} before calling.
+     * This method also read the number of values in the page from the header.
+     */
     private synchronized void ensurePageHeader(SeekableByteChannel file) throws IOException {
         if (pageHeader == null) {
-            offset = file.position();
+            if (file.position() != offset) {
+                throw new IllegalStateException("File position = " + file.position() + " not equal to expected offset ="
+                        + offset);
+            }
```

Comment:
This is silly. All of the callers invoke `readChannel.position(offset)` immediately before calling this method. Move that logic in here, and delete the check.
If you want to get fancier, you could read the current file position before hand and then restore it afterwards (right now you unconditionally restore it to `offset`).

Reply:
Its important to set the file position to offset because if we have already read the header, offset will store the offset for the data in the page. So we want to set file position as the location of data.